.changeset/strong-otters-brush.md

```diff
@@ -0,0 +1,5 @@
+---
+"single-spa-react": patch
```

Comment:
Whenever the `exports` field changes in the package.json, it might need to be a major change because of how different bundlers and NodeJS versions handle it.

```suggestion
"single-spa-react": major
```

Reply:
This fix should handle all resolution strategies (all typescript strategies), along with all popular bundlers. Still, if someone is using an obscure tool, it might behave differently, so we could treat it as a breaking change. Problem is not only with the typescript version, but a combination of typescript "moduleResolution", and bundlers. Currently, create-single-spa uses the "node" (which is actually node10) resolution strategy. What do you think?

I have inserted a compatibility layer that fixes the issue with certain bundlers requiring access to default export with ".default". Now, all of the resolution strategies are correctly handled:

![image](https://github.com/single-spa/single-spa-react/assets/9189985/d8fae0ce-b312-4061-9c91-2b358f25a692)

Reply:
Being totally honest, I would prefer calling it a major version just in case there's a weird edge case where it changes how things work. I don't know all the edge cases, but have had lots of github issues after changing my `exports` fields before.

I prefer single-spa-react users to face no risk when doing patch upgrades.

Reply:
Changed it to Major.